bundles/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/FreeboxBinding.java

```diff
@@ -222,42 +251,162 @@ protected void execute() {
 	@SuppressWarnings("incomplete-switch")
 	@Override
 	protected void internalReceiveCommand(String itemName, Command command) {
+		if ((loginManager == null) || (!loginManager.isConnected())) {
+			try {
+				authorize();
```

Comment:
since the authorisation is a long running process i would avoid calling it in receiveCommand. Better just check whether the Binding is authorised and in case it isn't return fast and log a reasonable statement explaining the problem.